SPEC.md

```diff
@@ -116,34 +119,62 @@ In this case, the app can reference the name "example.com/trusted-certificate-au
 The dependencies are applied in order and each image dependency can overwrite files from the previous dependency.
 An optional *path whitelist* can be provided, in which case all non-specified files from all dependencies will be omitted in the final, assembled rootfs.
 
-Image Format TODO
+## App Container Executor
 
-* Define the garbage collection lifecycle of the container filesystem including:
+The **App Container Executor** defines the process by which applications contained in ACIs are executed.
+There are two "perspectives" in this process.
+The "*executor*" perspective consists of the steps that the App Container Executor (ACE) must take to set up the environment for the pod and applications.
+The "*app*" perspective is how the app processes inside the pod see the environment.
+
+## Pods
+
+The deployable, executable unit in the App Container specification is the **pod**.
+A **pod** is a list of apps that should be launched together inside a shared context.
+The shared context is defined as the conjunction of the following Linux namespaces (or equivalents on other operating systems):
+- PID namespace
+- network namespace
+- mount namespace
+- IPC namespace
+- UTS namespace
+The context may also consist of one or more isolators.
+
+Each app in a pod will start chrooted into its own unique read-write rootfs before execution.
+
+The definition of the **pod** - namely, the list of constituent apps, and any isolators that should apply to the entire pod - is codified in a [Pod Manifest](#pod-manifest-schema).
+Pod Manifests can serve the role of both _deployable template_ and _runtime manifest_: a template can be a candidate for a series of transformations before execution.
+For example, a Pod Manifest might reference an app with a label requirement of `version=latest`, which another tool might subsequently resolve to a specific version.
+Another example would be that volumes are "late-bound" by the executor; alternatively, an executor might add annotations.
+
+A Pod Manifest must be fully resolved (_reified_) before execution.
+Specifically, a Pod Manifest must have all `mountPoint`s satisfied by `volume`s, and must reference all applications deterministically (by image ID).
+At runtime, the reified Pod manifest is exposed to applications through the metadata service.
+
+Pod TODO:
+* Define the garbage collection lifecycle of the pod filesystem including:
     * Format of app exit code and signal
     * The refcounting plan for resources consumed by the ACE such as volumes
-* Define the lifecycle of the container as all exit or first to exit
-* Define security requirements for a container. In particular is any isolation of users required between containers? What user does each application run under and can this be root (i.e. "real" root in the host).
-* Define how apps are supposed to communicate; can they/do they 'see' each other (a section in the apps perspective would help)?
+* Define the lifecycle of the pod as all exit or first to exit
```

Comment:
This makes sense from my perspective on things. For one, I think we can agree that "process still resident" isn't any real sort of health metric, so pod health and pod lifecycle management are likely to be, in effect, out of band from the executor (e.g. a concern of monitoring/scheduling system). Additionally, there are likely scenarios where a pod has several apps but one is expected to, for instance, do some work and terminate.